sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -112,6 +112,52 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
     )
   }
 
+  test("union by name") {
+    // Simple case
+    val df1 = Seq((1, "a", 3.0)).toDF("a", "b", "c")
+    val df2 = Seq((1.2, 2, "bc")).toDF("c", "a", "b")
+    val df3 = Seq(("def", 1.2, 3)).toDF("b", "c", "a")
+    val unionDf = df1.unionByName(df2.unionByName(df3))
+    checkAnswer(unionDf,
+      Row(1, "a", 3.0) :: Row(2, "bc", 1.2) :: Row(3, "def", 1.2) :: Nil
+    )
```

Comment:
Hi, @maropu .
To be clearer, could you add more test cases requiring type coercions here?